codalab/model/worker_model.py

```diff
@@ -316,65 +362,24 @@ def get_stream(self, sock, timeout_secs):
         except socket.timeout:
             return None
 
-    def get_json_message(self, sock, timeout_secs):
+    def recv_json_message_with_sock(self, sock, timeout_secs):
         """
         Receives a single message on the given socket and returns the message
         data parsed as JSON.
 
         If no messages are received within timeout_secs seconds, returns None.
         """
-        fileobj = self.get_stream(sock, timeout_secs)
+        fileobj = self.recv_stream(sock, timeout_secs)
 
         if fileobj is None:
             return None
 
         with closing(fileobj):
             return json.loads(fileobj.read().decode())
 
-    def send_stream(self, socket_id, fileobj, timeout_secs):
-        """
-        Streams the given file-like object to the given socket.
-
-        If nothing accepts a connection on the socket for more than timeout_secs,
-        return False. Otherwise, returns True.
-        """
-        start_time = time.time()
-        while time.time() - start_time < timeout_secs:
-            with closing(socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)) as sock:
-                sock.settimeout(timeout_secs)
-
-                success = False
-                try:
-                    sock.connect(self._socket_path(socket_id))
-                    success = sock.recv(len(WorkerModel.ACK)) == WorkerModel.ACK
-                except socket.error:
-                    logging.debug("socket error when calling send_stream")
-
-                if not success:
-                    # Shouldn't be too expensive just to keep retrying.
-                    time.sleep(0.003)
-                    continue
-
-                while True:
-                    data = fileobj.read(4096)
-                    if not data:
-                        return True
-                    sock.sendall(data)
-
-        return False
-
-    def _ping_worker_ws(self, worker_id):
-        async def ping_ws():
-            async with websockets.connect(f"{self._ws_server}/main") as websocket:
-                await websocket.send(worker_id)
-
-        futures = [ping_ws()]
-        loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(loop)
-        loop.run_until_complete(asyncio.wait(futures))
-        logging.warn(f"Pinged worker through websockets, worker id: {worker_id}")
-
-    def send_json_message(self, socket_id, worker_id, message, timeout_secs, autoretry=True):
+    def send_json_message_with_sock(
```

Comment:
Keep the same order as before so we minimize diffs

Reply:
Done.